core/src/main/java/hudson/XmlFile.java

```diff
@@ -246,12 +246,12 @@ public boolean exists() {
         return file.exists();
     }
 
-    public void delete() {
-        file.delete();
+    public void delete() throws IOException {
+        Files.deleteIfExists(Util.fileToPath(file));
     }
```

Comment:
Technically an API change that affects 7 plugins at compile-time, but I don't think the addition of the `throws` should affect those seven plugins at runtime. I think this is probably fine, but if people feel strongly otherwise I can convert this to an `UncheckedIOException` in order to preserve the existing method signature for those 7 plugins.

Reply:
Which plugins?

Reply:
@jglick Did you have any opinion about this part? This is what I'm the most unsure about, so would be good to hear if you think this is OK or if you'd prefer the `UncheckedIOException` approach here.

Reply:
Probably fine—may cause a compilation error on upgrade, but trivially resolved.

Reply:
copyartifact
    ecutest
    gerrit-trigger
    google-container-registry-auth
    jqs-monitoring
    nodejs
    nuget
    xvfb